DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/AddDelegateToGroupController.cs

```diff
@@ -0,0 +1,151 @@
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Helpers.FilterOptions;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.AddDelegateToGroup;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    [SetSelectedTab(nameof(NavMenuTab.Delegates))]
+    [Route("TrackingSystem/Delegates/Groups/AddDelegateToGroup")]
+    public class AddDelegateToGroupController : Controller
+    {
+        private const string AddDelegateToGroupCookieName = "AddDelegateToGroupFilter";
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IClockService clockService;
+        private readonly IGroupsDataService groupsDataService;
+        private readonly IGroupsService groupsService;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserDataService userDataService;
+
+        public AddDelegateToGroupController(
+            IJobGroupsDataService jobGroupsDataService,
+            IUserDataService userDataService,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IGroupsService groupsService,
+            IGroupsDataService groupsDataService,
+            IClockService clockService
+        )
+        {
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userDataService = userDataService;
+            this.groupsService = groupsService;
+            this.groupsDataService = groupsDataService;
+            this.clockService = clockService;
+        }
+
+        [Route("{groupId:int}")]
+        public IActionResult Index(
+            int groupId = 1,
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
+        {
+            sortBy ??= DefaultSortByOptions.Name.PropertyName;
+            filterBy = FilteringHelper.GetFilterBy(
+                filterBy,
+                filterValue,
+                Request,
+                AddDelegateToGroupCookieName,
+                DelegateActiveStatusFilterOptions.IsActive.FilterValue

```

Comment:
This is not a valid filter option for this feature. Filter options are:

-  Job Group
-  any Registration Prompts with set answers (where Centres.FXOptions is not null)

Reply:
This has now been refactored as per the original comment.